x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -128,3 +131,29 @@ function createFakeDatasource(pkg: RegistryPackage, assets: Asset[] = []): Datas
     ],
   };
 }
+
+async function ingestDatasourceCreate({
+  request,
+  datasource,
+}: {
+  request: Request;
+  datasource: Datasource;
+}) {
+  // OMG, so gross! Will not keep
+  // if we end up keeping the "make another HTTP request" method,
+  // we'll clean this up via proxy or something else which prevents these functions from needing to know this.
+  // The key here is to show the Saved Object we create being stored/retrieved by Inges
+  const base = `http://${request.headers.host}`;
+  const url = `${base}/api/ingest/datasources`;
+  const body = { datasource };
```

Comment:
The legacy request has `request.getBasePath()` which returns a correct base path, including the base path (if configured) and the space the user is currently using (if other than default). According to the migration documentation this is https://github.com/elastic/kibana/blob/master/docs/development/core/server/kibana-plugin-server.httpservicesetup.basepath.md in the new platform.

Either way, if we make an HTTP request here, we should use the correct helper method to make sure we have the correct base path.

Reply:
Added in https://github.com/elastic/kibana/pull/52964/commits/c07f1f710adba55eb61d6bfb390f7721fc2e43fb